### ISSUE TITLE
Add a search feature to search through certain columns in the database

### ISSUE THREAD

**Issue by osamaramihafez** [Quran-Journey/backend#142]
We need to be able to search for a surah based on a substring of the surah name.

Quran.com also allows you to search by surah number, page number, a surah:verse key, etc.

A good video on implementing search in postgres: https://www.youtube.com/watch?v=szfUbzsKvtE&

What ts_vectors are and how they help with full text search: https://forestry.io/blog/full-text-searching-with-postgres/#:~:text=The%20tsvector%20Type,of%20distinct%20words%20called%20lexemes.

**Comment by udokaokoye** [comment]
Do you have a remote database? or an export of your current database?

**Comment by osamaramihafez** [comment]
No, we currently just have the schema and a few scripts that populate the database with incomplete mock data.
